frontend/src/app/accountCreation/MfaTotp/MfaTotp.tsx

```diff
@@ -1,15 +1,43 @@
+import { useEffect, useState } from "react";
+import { Redirect } from "react-router";
+
 import { Card } from "../../commonComponents/Card/Card";
 import { CardBackground } from "../../commonComponents/CardBackground/CardBackground";
 import Button from "../../commonComponents/Button/Button";
 import StepIndicator from "../../commonComponents/StepIndicator";
 import { accountCreationSteps } from "../../../config/constants";
+import iconLoader from "../../../../node_modules/uswds/dist/img/loader.svg";
+import { LoadingCard } from "../LoadingCard/LoadingCard";
 
 interface Props {
-  qrCode: string;
+  enrollFunction: Function;
   totpType: string;
 }
 
 export const MfaTotp = (props: Props) => {
+  const [qrCode, setQrCode] = useState("");
+  const [submitted, setSubmitted] = useState(false);
+
+  useEffect(() => {
+    const getQrCode = async () => {
+      const { qrcode } = await props.enrollFunction();
```

Comment:
This `useEffect` runs once on component load to enroll the factor and grab the link to the QR code from the backend